### PR TITLE
protocol: Added states for introduce protocol

### PR DESCRIPTION
**Pull request by soluchok**
This branch name was renamed to see old comments visit this [link](https://github.com/hyperledger/aries-framework-go/pull/285)

Issue: https://github.com/hyperledger/aries-framework-go/issues/287

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/introduce/service.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	// Spec defines the introduce spec
+	Spec = metadata.AriesCommunityDID + ";spec/introduce/1.0/"
+	// ConnectionProposal defines the introduce proposal message type.
+	ConnectionProposal = Spec + "proposal"
+	// ConnectionResponse defines the introduce response message type.
+	ConnectionResponse = Spec + "response"
+	// ConnectionInvitation defines the introduce invitation message type.
+	ConnectionInvitation = Spec + "invitation"
+	// ConnectionConfirm defines the introduce confirm message type.
+	ConnectionConfirm = Spec + "confirm"
```
Comment (llorllale) [inline pkg/didcomm/protocol/introduce/service.go]: This is the only blocker for me - like I said [here](https://github.com/hyperledger/aries-framework-go/pull/285#discussion_r326180812), "confirm" is not a msg defined in the `introduce` RFC. We should use `ack`.

File: pkg/didcomm/protocol/introduce/service.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	// Introduce protocol name
+	Introduce = "introduce"
+	// Spec defines the introduce spec
+	Spec = metadata.AriesCommunityDID + ";spec/introduce/1.0/"
+	// ConnectionProposal defines the introduce proposal message type.
+	ConnectionProposal = Spec + "proposal"
+	// ConnectionResponse defines the introduce response message type.
+	ConnectionResponse = Spec + "response"
+	// ConnectionInvitation defines the introduce invitation message type.
+	ConnectionInvitation = Spec + "invitation"
```
Comment (llorllale) [inline pkg/didcomm/protocol/introduce/service.go]: `invitation` is also not an `introduce` message type. This is actually a `didexchange` msgType. What the Introducer does is it forwards the `Invitation`s to each introducee.

File: pkg/didcomm/protocol/introduce/service.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import "github.com/hyperledger/aries-framework-go/pkg/common/metadata"
+
+const (
+	// Introduce protocol name
+	Introduce = "introduce"
+	// Spec defines the introduce spec
+	Spec = metadata.AriesCommunityDID + ";spec/introduce/1.0/"
+	// ConnectionProposal defines the introduce proposal message type.
+	ConnectionProposal = Spec + "proposal"
```
Comment (rolsonquadras) [inline pkg/didcomm/protocol/introduce/service.go]: May be we can use "Introduce" prefix here as this is an introduce protocol.

**Comment by rolsonquadras** [comment]
@soluchok could you please link the issue# to this PR ?

File: pkg/didcomm/protocol/introduce/service.go (hunk 3 above)
Comment (soluchok) [reply pkg/didcomm/protocol/introduce/service.go]: I did that on purpose, my IDE (GoLand) says `Name starts with package name`

Comment (llorllale) [reply pkg/didcomm/protocol/introduce/service.go]: @soluchok I don't understand your comment? This is package `introduce`. Also, we're missing the [`request`](https://github.com/hyperledger/aries-rfcs/tree/master/features/0028-introduce#request) message.

**Pull request merged**